sql-cli/pyproject.toml

```diff
@@ -26,11 +26,51 @@ Home = "https://astronomer.io/"
 Source = "https://github.com/astronomer/astro-sdk/sql-cli"
 
 [tool.poetry.dependencies]
-python = ">=3.8,<4.0.0"
+python = ">=3.8,<3.11"
 typer = {extras = ["all"], version = "^0.6.1"}
 networkx = "^2.8.7"
 Jinja2 = ">=2.10.1, <=3.1"
 python-frontmatter = "^1.0.0"
+PyYAML = ">5.4.1"
+apache-airflow = ">2.0"
+python-dotenv = "^0.21.0"
+
+[tool.poetry.group.apache_airflow_providers_google]
+optional = true
+
+[tool.poetry.group.apache_airflow_providers_google.dependencies]
+protobuf = "<=3.20" # Google bigquery client require protobuf <= 3.20.0. We can remove the limitation when this limitation is removed
+apache-airflow-providers-google = ">=6.4.0"
+sqlalchemy-bigquery = { version = ">=1.3.0", python = ">=3.8,<3.11"}
+smart-open = {version = ">=5.2.1", extras = ["gcs"]}
+
+[tool.poetry.group.apache_airflow_providers_snowflake]
+optional = true
+
+[tool.poetry.group.apache_airflow_providers_snowflake.dependencies]
+apache-airflow-providers-snowflake = "*"
+snowflake-sqlalchemy = ">=1.2.0, <=1.2.4"
+snowflake-connector-python = {version = "*", extras=["pandas"]}
+
+[tool.poetry.group.apache_airflow_providers_postgres]
+optional = true
+
+[tool.poetry.group.apache_airflow_providers_postgres.dependencies]
+apache-airflow-providers-postgres = "*"
+
+[tool.poetry.group.apache_airflow_providers_amazon]
+optional = true
+
+[tool.poetry.group.apache_airflow_providers_amazon.dependencies]
+apache-airflow-providers-amazon = "*"
+s3fs = "*"
+smart-open = {extras = ["s3"], version = ">=5.2.1"}
+
+[tool.poetry.extras]
+google = ["protobuf", "apache-airflow-providers-google", "sqlalchemy-bigquery", "smart-open[gcs]"]
+snowflake = ["apache-airflow-providers-snowflake","snowflake-sqlalchemy", "snowflake-connector-python[pandas]"]
+postgres = ["apache-airflow-providers-postgres"]
+amazon = ["apache-airflow-providers-amazon", "s3fs", "smart-open[s3]"]
```

Comment:
This will be replaced by `astro-sdk[all]` at some point or? Do we have explicit need for these requirements in the connection validation?

Reply:
Removed all of it.